openhands/core/main.py

```diff
@@ -284,5 +284,6 @@ def generate_sid(config: AppConfig, session_name: str | None = None) -> str:
             config=config,
             initial_user_action=initial_user_action,
             sid=sid,
+            exit_on_message=True,
```

Comment:
NOTE: Avoid waiting for user input forever. Providing a fake user response is the standard solution though.